src/devices/Arduino/ArduinoBoard.cs

```diff
@@ -0,0 +1,177 @@
+using System.Device.Gpio;
+using System.Device.I2c;
+using System.Device.Pwm;
+using System.Device.Spi;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using Iot.Device.Spi;
+
+#pragma warning disable CS1591
+
+namespace Iot.Device.Arduino
+{
+    /// <summary>
+    /// Implements an interface to an arduino board which is running Firmata.
+    /// See documentation on how to prepare your arduino board to work with this.
+    /// Note that the program will run on the PC, so you cannot disconnect the
+    /// Arduino while this driver is connected.
+    /// </summary>
+    public class ArduinoBoard : IDisposable
+    {
+        private Stream _serialPortStream;
+        private FirmataDevice _firmata;
+        private Version _firmwareVersion;
+        private string _firmwareName;
+        private List<SupportedPinConfiguration> _supportedPinConfigurations;
+
+        public ArduinoBoard(Stream serialPortStream)
+        {
+            _serialPortStream = serialPortStream;
+        }
+
+        public event Action<string, Exception> LogMessages;
+
+        public virtual void Initialize()
+        {
+            _firmata = new FirmataDevice();
+            _firmata.Open(_serialPortStream);
+            _firmata.OnError += FirmataOnError;
+            var protocolVersion = _firmata.QueryFirmataVersion();
+            if (protocolVersion != _firmata.QuerySupportedFirmataVersion())
+            {
+                throw new NotSupportedException($"Firmata version on board is {protocolVersion}. Expected {_firmata.QuerySupportedFirmataVersion()}. They must be equal.");
+            }
+
+            Log($"Firmata version on board is {protocolVersion}.");
+
+            _firmwareVersion = _firmata.QueryFirmwareVersion(out _firmwareName);
+
+            Log($"Firmware version on board is {_firmwareVersion}");
+
+            _firmata.QueryCapabilities();
+
+            _supportedPinConfigurations = _firmata.PinConfigurations; // Clone reference
+
+            Log("Device capabilities: ");
+            foreach (var pin in _supportedPinConfigurations)
+            {
+                Log(pin.ToString());
+            }
+
+            // _firmata.SetSamplingInterval(TimeSpan.FromMilliseconds(100));
+            _firmata.EnableDigitalReporting();
+        }
+
+        public Version FirmwareVersion
+        {
+            get
+            {
+                return _firmwareVersion;
+            }
+        }
+
+        public string FirmwareName
+        {
+            get
+            {
+                return _firmwareName;
+            }
+        }
+
+        internal FirmataDevice Firmata
+        {
+            get
+            {
+                return _firmata;
+            }
+        }
+
+        internal List<SupportedPinConfiguration> SupportedPinConfigurations
+        {
+            get
+            {
+                return _supportedPinConfigurations;
+            }
+        }
+
+        internal void Log(string message)
```

Comment:
NIT: I know you are not using it, but perhaps you want this Log method to also have an exception parameter which is also passed in to the eventhandler

Reply:
True, I'll fix that.